ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx

```diff
@@ -0,0 +1,529 @@
+            logIngestionEndpoint: yup
+                .string()
+                .trim()
+                .required('A log ingestion endpoint is required'),
+            directoryTenantId: yup.string().trim().required('A directory tenant ID is required'),
+            applicationClientId: yup
+                .string()
+                .trim()
+                .required('A application client ID is required'),
+            secret: yup.string().trim(),
+            clientCertAuthConfig: yup.object().shape({
+                clientCert: yup.string().trim(),
+                privateKey: yup.string().trim(),
+            }),
+        }),
+    }),
+    updatePassword: yup.bool(),
+});
+
+export const defaultValues: MicrosoftSentinelFormValues = {
+    notifier: {
+        id: '',
+        name: '',
+        microsoftSentinel: {
+            logIngestionEndpoint: '',
+            directoryTenantId: '',
+            applicationClientId: '',
+            clientCertAuthConfig: {
+                clientCert: '',
+                privateKey: '',
+            },
+            secret: '',
+            alertDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+            auditLogDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+        },
+        labelDefault: '',
+        labelKey: '',
+        uiEndpoint: window.location.origin,
+        type: 'microsoftSentinel',
+    },
+    updatePassword: true,
+};
+
+function MicrosoftSentinelForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<MicrosoftSentinel>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    if (initialValues) {
+        formInitialValues.notifier = {
+            ...formInitialValues.notifier,
+            ...initialValues,
+        };
+        // We want to clear the password because backend returns '******' to represent that there
+        // are currently stored credentials
+        formInitialValues.notifier.microsoftSentinel.secret = '';
+        // Don't assume user wants to change password; that has caused confusing UX.
+        formInitialValues.updatePassword = false;
+    }
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onTest,
+        onCancel,
+        message,
+    } = useIntegrationForm<MicrosoftSentinelFormValues>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+    const { isCreating } = usePageState();
+    const [selectedAuthMethod, setSelectedAuthMethod] = useState(
+        values.notifier.microsoftSentinel.clientCertAuthConfig.clientCert
+            ? 'use-client-cert'
+            : 'use-secret'
+    );
```

Comment:
Van, am available for call if we need to discuss.

The hook is second source of truth for a value that can be derived directly from Formik state.

1. Replace with assignment that is evaluated on every render:
    ```ts
    const selectedAuthMethod = … ? 'use-client-cert' : 'use-secret`;
    ```
2. Move logic from `preOnSaveHook` into `onUpdateAuthMethod` function:

    ```ts
    // clear the previous authentication method
    if (selectedAuthMethod === 'use-client-cert') {
        setFieldValue('notifier.microsoftSentinel.secret', '');
    } else {
        setFieldValue('notifier.microsoftSentinel.clientCertAuthConfig.clientCert', '');
        setFieldValue('notifier.microsoftSentinel.clientCertAuthConfig.privateKey', '');
    }
    ```

Reply:
I actually had something similar, but Simon and I discussed and decided it was better to keep the UI state in memory. This prevents pollution the integration object with a value it doesn't store on the backend, and allows the user to explore both objects to see which one they would prefer to use.

Reply:
Okay, finally I get it.
1. The local state is how user chooses which credential. I was thinking originally from viewpoint after it has been set.
2. Inverse of what you experienced, I had to start this way for Quay robot accounts, but in that case, it turned out all 4 combinations (none, the one, the other, or both) of both credentials were needed in some scenario, so the form renders all of the inputs.
3. Therefore, because this is the only form that has this idiom, some comment might have been nice, for current reviewers and for future imitators within or outside of UI team of this as reference implementation.